modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -250,9 +254,30 @@ class SeqexecEngine(httpClient: Client[IO], settings: SeqexecEngine.Settings, sm
   ).getOrElse(st)
 
   def moveSequenceInQueue(q: EventQueue, qid: QueueId, seqId: Observation.Id, d: Int): IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
-    Event.modifyState[executeEngine.ConcreteTypes](moveSeq(qid, seqId, d) withEvent NullSeqEvent)
+    Event.modifyState[executeEngine.ConcreteTypes]((moveSeq(qid, seqId, d) withEvent NullSeqEvent).toHandle)
+  ).map(_.asRight)
+
+  private def runQueue(qid: QueueId, clientId: ClientID): executeEngine.HandleType[Unit] =
+    executeEngine.get.map(nextRunnableObservations(qid)).flatMap(_.map(executeEngine.start(_, clientId, {_ => true})).fold(executeEngine.unit)(_ *> _))
+
+  def startQueue(q: EventQueue, qid: QueueId, clientId: ClientID): IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
+    Event.modifyState[executeEngine.ConcreteTypes](executeEngine.get.flatMap{ st => {
+      (EngineState.queues ^|-? index(qid)).getOption(st).map {
+        _.status(st) match {
+          case BatchExecState.Idle     => ((EngineState.queues ^|-? index(qid) ^|-> ExecutionQueue.cmdState).set(BatchCommandState.Run(clientId)) >>> {(_, ())}).toHandle *> runQueue(qid, clientId)
+          case BatchExecState.Stopping => ((EngineState.queues ^|-? index(qid) ^|-> ExecutionQueue.cmdState).set(BatchCommandState.Run(clientId)) >>> {(_, ())}).toHandle
+          case _                       => executeEngine.unit
+        }
+      }.getOrElse(executeEngine.unit)
+    }}.map(_ => StartQueue(qid, clientId)))
   ).map(_.asRight)
 
+  // It assumes only one queue can run at a time
+  private val iterateQueues: PartialFunction[SystemEvent, executeEngine.HandleType[Unit]] = {
```

Comment:
This is the function that reacts to sequence completion, to start the next sequences in the queue.